bin/start.js

```diff
@@ -43,50 +40,66 @@ if (semver.lt(process.versions.node, '4.0.0')) {
   process.exit(1)
 }
 
-var knownOpts = {
-  help: Boolean,
-  version: Boolean,
-  loglevel: [
-    'silly',
-    'verbose',
-    'info',
-    'http',
-    'warn',
-    'error',
-    'silent'
-  ],
-  port: Number,
-  'bind-address': String,
-  public: path,
-  'in-memory': Boolean,
-  data: path,
-  'db-url': String
-}
-
-var shortHands = {
-  h: '--help',
-  usage: '--help',
-  v: '--version',
-  m: '--in-memory',
-  s: ['--loglevel', 'silent'],
-  d: ['--loglevel', 'info'],
-  dd: ['--loglevel', 'verbose'],
-  ddd: ['--loglevel', 'silly'],
-  silent: ['--loglevel', 'silent'],
-  verbose: ['--loglevel', 'verbose'],
-  quiet: ['--loglevel', 'warn']
-}
-
-var argv = nopt(knownOpts, shortHands)
-
-if (argv.help) {
-  process.stdout.write(fs.readFileSync(path.join(__dirname, 'readme.txt'), 'utf8'))
-  process.exit(0)
-}
-
-if (argv.version) {
+var args = yargs
+.options({
+  loglevel: {
+    choices: [
+      'silly',
+      'verbose',
+      'info',
+      'http',
+      'warn',
+      'error',
+      'silent'
+    ],
+    default: function () {
+      if ('s' in yargs.argv || 'ddd' in yargs.argv) return 'silly'
+      if ('d' in yargs.argv) return 'info'
+      if ('dd' in yargs.argv || 'verbose' in yargs.argv) return 'verbose'
+      if ('silent' in yargs.argv) return 'silent'
+      return 'warn'
+    }
+  },
+  port: {
+    type: 'number',
+    default: 6004
+  },
+  'bind-address': {
+    type: 'string',
+    default: '0.0.0.0'
+  },
+  public: {
+    type: 'string',
+    default: undefined,
```

Comment:
I think it would be good to say that default values for `--public` and `--data` are `./public` and `./.hoodie` respectively. We could also add descriptions to them, as well as `--in-memory` and `--db-url`, see https://github.com/hoodiehq/hoodie#usage.

Sorry that I only bring this up now